src/core/utility/rate.cpp

```diff
@@ -61,7 +61,7 @@ void CRate::Sleep() {
 
 void CRate::SetRate(Real f_rate) {
   m_fNominalRate = Abs(f_rate);
-  m_unNominalPeriod = 1e6 / m_fNominalRate;
+  m_unNominalPeriod = static_cast<UInt32>(1e6 / m_fNominalRate);
```

Comment:
Are these casts really necessary? Is it a warning that makes them necessary?

Reply:
Same thing here